### PR TITLE
Sync much of the packaging improvements done in Debian.org to upstream MariaDB.org

### PR DESCRIPTION
**Pull request by ottok**

### CODE REVIEW

**Comment by ottok** [comment]
On commit updates the Travis-CI config, so here is verification that it was correct:
image

**Comment by ottok** [comment]
Fun fact: the new Travis-CI 10000 credits max open source limit is enough to do one MariaDB build :+1: 
image

**Comment by ottok** [comment]
Buildbot failure on updated dependencies is intentional:
http://buildbot.askmonty.org/buildbot/grid?branch=ok-10.5-deb-sync&category=main&category=experimental
```
 === /usr/lib/x86_64-linux-gnu/libmariadb3/plugin/auth_gssapi_client.so
 === /usr/lib/x86_64-linux-gnu/libmariadb3/plugin/auth_test_plugin.so
 === /usr/lib/x86_64-linux-gnu/libmariadb3/plugin/caching_sha2_password.so
 === /usr/lib/x86_64-linux-gnu/libmariadb3/plugin/client_ed25519.so
 === /usr/lib/x86_64-linux-gnu/libmariadb3/plugin/dialog.so
 === /usr/lib/x86_64-linux-gnu/libmariadb3/plugin/mysql_clear_password.so
 === /usr/lib/x86_64-linux-gnu/libmariadb3/plugin/qa_auth_client.so
 === /usr/lib/x86_64-linux-gnu/libmariadb3/plugin/qa_auth_interface.so
 === /usr/lib/x86_64-linux-gnu/libmariadb3/plugin/sha256_password.so
+=== /usr/lib/x86_64-linux-gnu/libmariadb3/plugin/test_sql_service.so
+	/lib64/ld-linux-x86-64.so.2 
+	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 
+	linux-vdso.so.1 
++ [[ 1 -ne 0 ]]
++ echo 'ERROR: something has changed in the dependencies of binaries or libraries. See the diff above'
ERROR: something has changed in the dependencies of binaries or libraries. See the diff above
```

**Comment by ottok** [comment]
PR https://github.com/MariaDB/server/pull/1729 needs to be merged before this one, as that PR fixes Salsa-CI detectable regressions and after that this and all other PRs would be all green.

**Comment by ottok** [comment]
# Quality assurance

## Travis-CI

Passed OK before and after.

## Salsa-CI

One more test passes now after this PR:

https://salsa.debian.org/mariadb-team/mariadb-server/-/pipelines/213105
image

## Old buildbot

Various failures but mainly the same as on main 10.5 branch, not very usable as a regression test tool right now, too much effort in checking unrelated issues compared to real issues that might be caused by this PR: http://buildbot.askmonty.org/buildbot/grid?branch=ok-10.5-deb-sync&category=main&category=experimental

## New buildbot


Passed OK before and after.

File: debian/mariadb-server-10.5.README.Debian [hunk 1]
```diff
@@ -1,41 +1,145 @@
...
+file as is has been obsoleted.
+
+
+* MARIADB IS SECURE BY DEFAULT
+==============================
+
+MariaDB in Debian is secure by default, because:
+
+- It only listens to the localhost socket and cannot be accessed remotely unless
+  the sysadmin changes the configurationin /etc/mysql to allow so.
```
Comment (grooverdan) [inline debian/mariadb-server-10.5.README.Debian:82]: typo configuration

I wrote this doc too that may be useful to reference: https://mariadb.com/kb/en/authentication-plugin-unix-socket/#security

**Pull request merged**